src/congestion/BufferOccupancy.h

```diff
@@ -58,17 +58,15 @@ class BufferOccupancy : public CongestionSensor {
   void performDecrementCredit(u32 _vcIdx);
   void performDecrementWindow(u32 _vcIdx);
 
-  f64 vcStatusNorm(u32 _outputPort, u32 _outputVc) const;
-  f64 vcStatusAbs(u32 _outputPort, u32 _outputVc) const;
-  f64 portAverageStatusNorm(u32 _outputPort) const;
-  f64 portAverageStatusAbs(u32 _outputPort) const;
+  f64 vcStatus(u32 _outputPort, u32 _outputVc, bool _normalize) const;
+  f64 portAverageStatus(u32 _outputPort, bool _normalize) const;
 
   const u32 latency_;
   const Mode mode_;
 
-  std::vector<u32> creditMaximums_;
-  std::vector<u32> creditCounts_;
-  std::vector<u32> flitsOutstanding_;
+  // 64-bit to hold U32_MAX
```

Comment:
why? it seems to me there's nothing special about U32_MAX for outstandingFlits_. And if the queue is infinite, it seems to me that normalizationDivisors_ should be used because normalized mode on an infinite queue doesn't make sense, right?

Reply:
What if someone makes a 3e9 sized queue?

Reply:
Nothing wrong with s64, I just didn't understand the justification. If it's because there could be >= 2^31 flits outstanding on a VC, that's fine (but I would be surprised if something else didn't break in supersim before that)